src/librbd/internal.cc

```diff
@@ -1940,10 +1940,31 @@ bool compare_by_name(const child_info_t& c1, const child_info_t& c2)
     uint64_t period = src->get_stripe_period();
     unsigned fadvise_flags = LIBRADOS_OP_FLAG_FADVISE_SEQUENTIAL |
 			     LIBRADOS_OP_FLAG_FADVISE_NOCACHE;
+    uint64_t object_id = 0;
     for (uint64_t offset = 0; offset < src_size; offset += period) {
       if (throttle.pending_error()) {
         return throttle.wait_for_ret();
       }
+      
+      {
+        RWLock::RLocker snap_locker(src->snap_lock);
+        bool skip = true;
+        if (src->object_map != nullptr) {
+          // each period is related to src->stripe_count objects, check them all
+          for (uint64_t i=0; i < src->stripe_count; i++) {
+            if (object_id < src->object_map->size() &&
+                src->object_map->object_may_exist(object_id)) {
+              skip = false;
```

Comment:
Nit: you can break out of the loop

Reply:
break out of the loop may lead to object_id mismatch with period.
so to be simplicity, i choose to just let object_id increase at each step.